### DISCUSSION TITLE
cockpit patternfly setup in custom project

### DISCUSSION THREAD

**Discussion by Tobi-De** [cockpit-project/cockpit#20740] [Q&A] [answered]
Hi everyone,

I'm currently trying to build a Cockpit application. I'm not using the starter kit (yes, I know, already off to a bad start). My main question is, where are these two imports coming from?

```javascript
import "cockpit-dark-theme";
import "patternfly/patternfly-5-cockpit.scss";
```

I was able to add PatternFly to my project, and it works without issue, but I noticed the colors were off compared to the rest of Cockpit. I checked the starter kit and saw these two imports, but there is no corresponding package in the `package.json`. So, where am I supposed to get these from?

**Comment by martinpitt** [comment, marked as answer]
These come from cockpit's pkg/lib, which has shared components between all cockpit pages. starter-kit [checks them out in its Makefile](https://github.com/cockpit-project/starter-kit/blob/354f5523f53ce24d4fc643678c24c1d4657413cb/Makefile#L26).

**Reply by martinpitt** [reply]
Plus, starter-kit's `./build.js` adds `pkg/lib` to esbuild's `nodePaths`. Other bundlers have a similar config where to look for imports.
